Loop/Managers/LoopDataManager.swift

```diff
@@ -1896,23 +1842,32 @@ extension LoopDataManager {
         guard FeatureFlags.simulatedCoreDataEnabled else {
             fatalError("\(#function) should be invoked only when simulated core data is enabled")
         }
+        
+        guard let doseStore = doseStore as? DoseStore else {
+            fatalError("\(#function) should be invoked only when not using mock stores")
+        }
 
-        self.doseStore.purgeHistoricalPumpEvents() { error in
+        doseStore.purgeHistoricalPumpEvents() { error in
             guard error == nil else {
                 completion(error)
                 return
             }
             self.dosingDecisionStore.purgeHistoricalDosingDecisionObjects() { error in
-                guard error == nil else {
+                guard error == nil, let carbStore = self.carbStore as? CarbStore else {
```

Comment:
[nit] Relative to the preceding comment, why is carbStore handled specially? That is, no `fatalError` here? Consider previous recommendation of following pattern in `generateSimulatedHistoricalCoreData`.